pkg/service/mockrecord/mockrecord.go

```diff
@@ -52,6 +52,9 @@ func (s *mockRecorder) MockRecord(path string, pid uint32, mockName string) {
 		return
 	}
 
+	var proxyPortNumber proxy.Option;
+	proxyPortNumber.Port = proxyPort;
+
 	// start the proxy
 	ps := proxy.BootProxy(s.logger, proxy.Option{}, "", "", pid, "", []uint{}, loadedHooks, ctx)
```

Comment:
You didn't pass the `proxyPortNumber` here and ideally, you shouldn't, You should rather pass the `proxy.Option{Port:proxyPort}`